analysis/furtherModelTesting.ipynb

```diff
@@ -0,0 +1,1039 @@
+      "text/plain": [
+       "<Figure size 720x432 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "MAE for n_neighbors=1: 0.5410108892598948\n",
+      "MAE for n_neighbors=2: 0.5426444457645021\n",
+      "MAE for n_neighbors=3: 0.5513075877791929\n",
+      "MAE for n_neighbors=4: 0.573686912528912\n",
+      "MAE for n_neighbors=5: 0.5495526073067393\n",
+      "MAE for n_neighbors=6: 0.5634956319442919\n",
+      "MAE for n_neighbors=7: 0.5732355218202718\n",
+      "MAE for n_neighbors=8: 0.5738835808214869\n",
+      "MAE for n_neighbors=9: 0.571293157667966\n",
+      "MAE for n_neighbors=10: 0.5607889033389646\n",
+      "MAE for n_neighbors=11: 0.5448196712094987\n",
+      "MAE for n_neighbors=12: 0.5364807313533111\n",
+      "MAE for n_neighbors=13: 0.5280962038800316\n",
+      "MAE for n_neighbors=14: 0.5206492283871483\n",
+      "MAE for n_neighbors=15: 0.50927465617411\n",
+      "MAE for n_neighbors=16: 0.5081548192759396\n",
+      "MAE for n_neighbors=17: 0.5028496564392458\n",
+      "MAE for n_neighbors=18: 0.4968811868517823\n",
+      "MAE for n_neighbors=19: 0.4902695300407429\n",
+      "MAE for n_neighbors=20: 0.4865336079257342\n",
+      "RMSE for KNN is: 1.7631053829435306\n",
+      "MAE for KNN is: 0.9387104717208447\n"
+     ]
+    }
+   ],
+   "source": [
+    "#HYPERPARAMTER TUNING FOR KNN\n",
+    "from sklearn.model_selection import GridSearchCV\n",
+    "\n",
+    "#split into training and test sets\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=global_test_size, random_state=global_random_state)\n",
+    "\n",
+    "#train the model with KNN\n",
+    "#knn_model = KNeighborsRegressor(n_neighbors=5)\n",
+    "# Define the parameter grid\n",
+    "param_grid = {'n_neighbors': range(1, 21)}\n",
```

Comment:
range does not need to be so large. This slows down the computation and from the graph it's easy to see that anything past like 15 n_neighbours has a worse effect.

However, I believe this is proportional to the size of our data. Once Helios is finished, we will need to update ALL our parameters to match the new data we will be receiving.